BezierKit/Library/AugmentedGraph.swift

```diff
@@ -55,19 +55,35 @@ private class Node {
         self.forwardEdge = nil
         self.backwardEdge = nil
     }
+    var allIncidentEdges: [(Edge, Bool)] {
+        var result: [(Edge, Bool)] = []
+        func callback(_ node: Node) {
+            if let forward = node.forwardEdge { result.append((forward, true)) }
+            if let backward = node.backwardEdge { result.append((backward, false)) }
+        }
+        callback(self)
+        neighbors.forEach { callback($0) }
+        return result
+    }
```

Comment:
we could rephrase this as `enumerateIncidentEdges` with a callback to reduce array overhead.